src/service_identity/hazmat.py

```diff
@@ -50,6 +50,11 @@ def verify_service_identity(
     *obligatory_ids* must be both present and match.  *optional_ids* must match
     if a pattern of the respective type is present.
     """
+    if not cert_patterns:
```

Comment:
I guess that this is hazmat, so kind of private.
The API has no certificate as input.

But as long as the high level API raises this exception, it should be fine.

Thanks

Reply:
I bubbles up, but I've added an integration test for cryptography, too.